firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/chunks.txt

```diff
@@ -96,10 +96,16 @@ Scenario: chunksplitter correctly handles delimeters other than 0x00
 Scenario: Chunksplitter correctly handles input of multiple delimited chunks before output consumes them
 
   GIVEN("A uint8_t chunksplitter with an empty internal buffer of capacity 256 bytes")
-    WHEN("input to the chunksplitter is '0x01 0x02 0x03 0x00 0x04 0x05 0x00' ")
+    WHEN("input to the chunksplitter is '0x01 0x02 0x03 0x00 0x04 0x05 0x00' and output is called between the delimiter chunks")
       THEN("The input method reports output_ready status")
       THEN("the output method reports ok status")
-      THEN("The output buffer is as expected") // should not push subsequent bytes after the first delimeter to the buffer
+      THEN("The output buffer is as expected '\x04\x05' ")
+
+    WHEN("A delimiter byte input is sandwiched between two non-delimiter byte inputs")
+      THEN("The input method reports ok status") // 1st non-delimeter byte
+      THEN("The input method reports output_ready status") // delimeter byte
+      THEN("The input method reports ok status") // 2nd non-delimeter byte
```

Comment:
"The input method reports ok status and sets the input_overwritten flag on the 3rd byte"